ymer/src/main/java/com/avanza/ymer/YmerSpaceSynchronizationEndpoint.java

```diff
@@ -79,6 +90,12 @@ public void setApplicationContext(@Nonnull ApplicationContext applicationContext
 	@Override
 	public void onApplicationEvent(@Nonnull ContextRefreshedEvent event) {
 		if (event.getApplicationContext().equals(applicationContext)) {
+			try {
+				// This value should be available once the space is started
+				currentNumberOfPartitions = persistedInstanceIdRecalculationService.determineNumberOfPartitions();
```

Comment:
We might want to look at if we can get this value to be available earlier on in the startup process (before `onOperationsBatchSynchronization` is available to be called).

Reply:
Maybe extract it in `setApplicationContext` instead? That will be called earlier.

Reply:
That does not work in at least the integration tests. The space property isn't set in the context yet when `setApplicationContext` is called (it is set by GigaSpaces later on in the startup process).

Reply:
I resolved this by reading currentNumberOfPartitions from system property when setting up this class, and if not available as system property it will use the property from the spring context on context refresh, as it was done earlier.

If the system property is set, this should make `currentNumberOfPartitions` always be set when `onOperationsBatchSynchronization` might be called.